src/native/eventpipe/ds-sources.c

```diff
@@ -16,6 +16,52 @@
 #include "ds-protocol.c"
 #endif
 
+#undef PORTABLE_RID_ARCH
+
+#if defined(TARGET_X86)
+#define PORTABLE_RID_ARCH "x86"
+#elif defined(TARGET_AMD64)
+#define PORTABLE_RID_ARCH "x64";
+#elif defined(TARGET_ARMV6)
+#define PORTABLE_RID_ARCH "arm32";
+#elif defined(TARGET_ARM)
+#define PORTABLE_RID_ARCH "arm32";
+#elif defined(TARGET_ARM64)
+#define PORTABLE_RID_ARCH "arm64";
+#elif defined(TARGET_S390X)
+#define PORTABLE_RID_ARCH "s390x";
+#elif defined(TARGET_LOONGARCH64)
+#define PORTABLE_RID_ARCH "loongarch64";
+#elif defined(TARGET_POWERPC64)
+#define PORTABLE_RID_ARCH "ppc64le";
+#endif
+
+#undef PORTABLE_RID_OS
+
+#if defined(TARGET_UNIX)
+
+#if defined(TARGET_ANDROID)
+#define PORTABLE_RID_OS "linux-bionic"
+#elif defined(TARGET_LINUX_MUSL)
+#define PORTABLE_RID_OS "linux-musl"
+#elif defined(TARGET_LINUX)
+#define PORTABLE_RID_OS "linux"
+#elif defined(TARGET_OSX)
+#define PORTABLE_RID_OS "osx"
+#else
+#define PORTABLE_RID_OS "unix"
+#endif
+
+#elif defined(TARGET_WINDOWS)
+#define PORTABLE_RID_OS "win"
+#endif
+
+#if defined(PORTABLE_RID_OS) && defined(PORTABLE_RID_ARCH)
+const ep_char8_t* _ds_portable_rid_info = PORTABLE_RID_OS "-" PORTABLE_RID_ARCH;
+#else
+const ep_char8_t* _ds_portable_rid_info = "unknown";
```

Comment:
I think it would be better to get a build error if the ifdefs above do not cover the OS or architecture.

Reply:
Fixed